### PR TITLE
Update our tokenfactory to the latest version NTRN-93

### PR DESCRIPTION
**Pull request by quasisamurai**
[Task
](https://hadronlabs.atlassian.net/browse/NTRN-93)

- [Neutron PR](https://github.com/neutron-org/neutron/pull/307)
- [Intergration tests ](https://github.com/neutron-org/neutron-integration-tests/pull/200)
- [Neutron SDK](https://github.com/neutron-org/neutron-sdk/pull/110)
- [Dev contracts](https://github.com/neutron-org/neutron-dev-contracts/pull/26)

### CODE REVIEW

File: src/testcases/parallel/governance.test.ts [hunk 1]
```diff
@@ -389,6 +389,15 @@ describe('Neutron / Governance', () => {
       );
     });
 
+    test('create proposal #20, will pass', async () => {
+      await daoMember1.submitUpdateParamsInterchaintxsProposal(
+        'Proposal #20',
+        'Update interchaintxs params',
```
Comment (sotnikov-s) [inline src/testcases/parallel/governance.test.ts:395]: what about other modules? I'd like to make sure we didn't miss a thing in the copy+paste exercise

**Review by sotnikov-s — CHANGES_REQUESTED** [review]

File: src/testcases/parallel/governance.test.ts (hunk 1 above)
Comment (quasisamurai) [reply src/testcases/parallel/governance.test.ts:395]: we discussed it on refinement

**Comment by quasisamurai** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/6173587028/job/16762188644

File: src/testcases/parallel/tokenfactory.test.ts [hunk 2]
```diff
@@ -289,6 +320,25 @@ describe('Neutron / Tokenfactory', () => {
...
+        }),
+      );
+      const res = await neutronChain.queryContract<{
+        cosm_wasm_addr: string;
+      }>(contractAddress, {
+        before_send_hook: {
+          denom,
+        },
+      });
+      expect(res.cosm_wasm_addr).toEqual(contractAddress);
```
[the remark this answers is not in the document]
Comment (quasisamurai) [reply src/testcases/parallel/tokenfactory.test.ts]: I'd like too, but it's gonna require implementation of the whole contract w custom logic (and we don't want to waste any more time). so:

1. code of whole module is copypasted
2. module integrated successfully
3. the test will be more about contract written for this test
4. time pressure

File: src/testcases/parallel/governance.test.ts [hunk 3]
```diff
@@ -1035,6 +1044,29 @@ describe('Neutron / Governance', () => {
...
+    });
+  });
+
+  describe('try to execute proposal #20', () => {
+    const proposalId = 20;
+    test('check if proposal is passed', async () => {
+      await dao.checkPassedProposal(proposalId);
+    });
+    test('execute passed proposal', async () => {
+      await daoMember1.executeProposalWithAttempts(proposalId);
```
[the remark this answers is not in the document]
Comment (quasisamurai) [reply src/testcases/parallel/governance.test.ts:1066]: done

**Comment by quasisamurai** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/6173587028

**Comment by quasisamurai** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/6173587028

File: src/testcases/parallel/tokenfactory.test.ts [hunk 4]
```diff
@@ -187,6 +190,114 @@ describe('Neutron / Tokenfactory', () => {
...
+        newTokenDenom,
+      );
+      expect(contractBalance).toEqual(666);
+
+      let queryBlock = await neutronChain.queryContract<{
+        block: { received: boolean };
+      }>(contractAddress, {
+        sudo_result_block_before: {},
+      });
+      console.log(queryBlock);
```
Comment (NeverHappened) [inline src/testcases/parallel/tokenfactory.test.ts:247]: console.log

File: src/testcases/parallel/tokenfactory.test.ts [hunk 5]
```diff
@@ -187,6 +190,114 @@ describe('Neutron / Tokenfactory', () => {
...
+
+      let queryTrack = await neutronChain.queryContract<{
+        track: { received: boolean };
+      }>(contractAddress, {
+        sudo_result_track_before: {},
+      });
+      console.log(queryTrack);
```
Comment (NeverHappened) [inline src/testcases/parallel/tokenfactory.test.ts:254]: console.log

File: src/testcases/parallel/tokenfactory.test.ts [hunk 6]
```diff
@@ -187,6 +190,114 @@ describe('Neutron / Tokenfactory', () => {
...
+        denom: newTokenDenom,
+      });
+
+      const contractBalance = await neutronChain.queryDenomBalance(
+        contractAddress,
+        newTokenDenom,
+      );
+      expect(contractBalance).toEqual(666);
+
+      let queryBlock = await neutronChain.queryContract<{
```
Comment (NeverHappened) [inline src/testcases/parallel/tokenfactory.test.ts:242]: `queryBlock` is never checked actually

File: src/testcases/parallel/tokenfactory.test.ts [hunk 7]
```diff
@@ -187,6 +190,114 @@ describe('Neutron / Tokenfactory', () => {
...
+        block: { received: boolean };
+      }>(contractAddress, {
+        sudo_result_block_before: {},
+      });
+      console.log(queryBlock);
+
+      let queryTrack = await neutronChain.queryContract<{
```
Comment (NeverHappened) [inline src/testcases/parallel/tokenfactory.test.ts:249]: same: please check what's there

**Review by NeverHappened — CHANGES_REQUESTED** [review]

**Comment by quasisamurai** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/6173587028/job/16939576523 actually test are passing

**Comment by NeverHappened** [comment]
yeah, I have last tests passing now https://github.com/neutron-org/neutron-tests/actions/runs/6238607927

**Review by sotnikov-s — APPROVED** [review]

**Pull request merged**